src/bot/scripts/card.js

```diff
@@ -728,946 +66,1233 @@ const card = new Command({
   preload: true,
 
   // -card <name>
-  cb: async function (client, message, guild) {
-
-    // deep copy cards so we can replace shit
-    // without worrying about gay object shit
-    const cardsCopy = JSON.parse(JSON.stringify(cards));
-
-    // render frames for the website
-    if (config.env != null &&
-      config.env.toLowerCase() === "dev" &&
-      message.content.split(" ")[1] === "dev-frames") {
-
-      renderFrames(cardsCopy);
-      return;
-    }
-
-    // legacy command support
-
-    // card <name>
-    // card <name> <level>
-    // card <name> l<level>
-    // card <name> u<upgrade>
-
-    let name = null;
-
-    let level = null;
-    let upgrade = null;
-
-    // 5 = card(4) + space(1).
-    const args = message.content.slice(guild.prefix.length + 5);
-    if (args === "") {
+  cb: async function (client, message) {
+    if (Date.now() - new Date(cache.date) > hourInMs) {
+      await rp("https://sppd.feinwaru.com/api/v1/cards/list")
+        .then(async responseCache => {
+          const requestCache = JSON.parse(responseCache);
+
+          cache.date = new Date();
+          cache.cachedData = requestCache;
+        })
 
-      return message.reply(`usage: ${guild.prefix}card <name> (l)[level=1]/u[upgrade]`);
+        .catch(error => {
+          console.error(error);
+          const discordEmbed = new discord.RichEmbed()
+            .setColor("#F8534F")
+            .setTitle(":x: Error fetching data from cards/list");
+          return message.channel.send(discordEmbed);
+        });
     }
 
-    const split = args.split(" ");
-
-    // ***WARNING*** - this does change later in the code as a workaround,
-    // good luck to all the poor souls trying to debug this cancer
-    let last = split[split.length - 1];
+    rp("https://sppd.feinwaru.com/api/v1/cards/list")
```

Comment:
youre calling the route that youre supposed to cache every time the script is run :/

Reply:
uhh??

Reply:
youre not caching this request for an hour, youre supposed to